test/jdk/java/awt/FileDialog/FileDialogForPackages.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2013, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.FileDialog;
+import java.awt.Frame;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+
+/*
+ * @test
+ * @bug 8026869
+ * @summary Support apple.awt.use-file-dialog-packages property.
+ * @requires (os.family == "mac")
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FileDialogForPackages
+*/
+
+public class FileDialogForPackages {
+    private static final String APPLICATIONS_FOLDER = "/Applications";
+
+    private static volatile Button showBtn;
+    private static volatile FileDialog fd;
+
+    private static JFrame initialize() {
+        System.setProperty("apple.awt.use-file-dialog-packages", "true");
+
+        JFrame frame = new JFrame("Packages File Dialog Test Frame");
+        frame.setLayout(new BorderLayout());
+        JTextArea textOutput = new JTextArea(8, 30);
+        textOutput.setLineWrap(true);
+        JScrollPane textScrollPane = new JScrollPane(textOutput);
+        frame.add(textScrollPane, BorderLayout.CENTER);
+
+        fd = new FileDialog(new Frame(), "Open");
```

Comment:
Actually you don't need the textOutput, with recent changes you can add `.logArea()` to the PassFailJFrame builder

https://github.com/openjdk/jdk/blob/master/test/jdk/java/awt/regtesthelpers/PassFailJFrame.java#L1200

and just use `PassFailJFrame.log()` wherever you need it.

So this method can be changed to return the a `JButton`, and use it with `splitUIBottom`/`splitUIRight` instead of testUI

Reply:
Ok, using the PassFailJFrame logger.